File: bindings/lua/Polycode/BezierPoint.lua
class "BezierPoint"


function BezierPoint:__getvar(name)
	if name == "p1" then
		local retVal = Polycode.BezierPoint_get_p1(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "p2" then
		local retVal = Polycode.BezierPoint_get_p2(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "p3" then
		local retVal = Polycode.BezierPoint_get_p3(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function BezierPoint:__setvar(name,value)
	if name == "p1" then
		Polycode.BezierPoint_set_p1(self.__ptr, value.__ptr)
		return true
	elseif name == "p2" then
		Polycode.BezierPoint_set_p2(self.__ptr, value.__ptr)
		return true
	elseif name == "p3" then
		Polycode.BezierPoint_set_p3(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function BezierPoint:BezierPoint(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.BezierPoint(unpack(arg))
	end
end

function BezierPoint:__delete()
	if self then Polycode.delete_BezierPoint(self.__ptr) end
end

File: bindings/javascript/Polycode/BezierPoint.js
function BezierPoint(p1x,p1y,p1z,p2x,p2y,p2z,p3x,p3y,p3z) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.BezierPoint(p1x,p1y,p1z,p2x,p2y,p2z,p3x,p3y,p3z)
	}
	Object.defineProperties(this, {
		'p1': { enumerable: true, configurable: true, get: BezierPoint.prototype.__get_p1, set: BezierPoint.prototype.__set_p1},
		'p2': { enumerable: true, configurable: true, get: BezierPoint.prototype.__get_p2, set: BezierPoint.prototype.__set_p2},
		'p3': { enumerable: true, configurable: true, get: BezierPoint.prototype.__get_p3, set: BezierPoint.prototype.__set_p3}
	})
}

BezierPoint.prototype.__get_p1 = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.BezierPoint__get_p1(this.__ptr)
	return retVal
}

BezierPoint.prototype.__set_p1 = function(val) {
	Polycode.BezierPoint__set_p1(this.__ptr, val.__ptr)
}

BezierPoint.prototype.__get_p2 = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.BezierPoint__get_p2(this.__ptr)
	return retVal
}

BezierPoint.prototype.__set_p2 = function(val) {
	Polycode.BezierPoint__set_p2(this.__ptr, val.__ptr)
}

BezierPoint.prototype.__get_p3 = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.BezierPoint__get_p3(this.__ptr)
	return retVal
}

BezierPoint.prototype.__set_p3 = function(val) {
	Polycode.BezierPoint__set_p3(this.__ptr, val.__ptr)
}

Duktape.fin(BezierPoint.prototype, function (x) {
	if (x === BezierPoint.prototype) {
		return;
	}
	Polycode.BezierPoint__delete(x.__ptr)
})
